### ISSUE TITLE
problem with paths longer than 260 on windows

### ISSUE THREAD

**Issue by albfan** [ggreer/the_silver_searcher#712]
searching on long paths on windows gets:

    ERR: Error opening file ./whatever/long/path: No such file or directory Skipping...

I guess it is related to:

https://msdn.microsoft.com/en-us/library/windows/desktop/aa364963(v=vs.85).aspx

I'm talking with kkowalczyk@gmail.com, the creator of (ag windows port)

https://blog.kowalczyk.info/software/the-silver-searcher-for-windows.html

as this seems an easy fix.

Maybe more clues can be found on similar project

https://github.com/monochromegane/the_platinum_searcher

which work well with long paths but has less features than ag.

**Comment by mitchblank** [comment]
Basically you need to convert the path to be absolute, make sure it's normalized to `\` path separators and then prefix it with `\\?\`... unless it was originally a UNC path in which case you have to change the initial `\\` to `\\?\UNC\`.  This is all described in "Maximum Path Length Limitation" here https://msdn.microsoft.com/en-us/library/aa365247(VS.85).aspx

I think you need to also make sure to only use fs APIs targetting the native unicode interfaces, but I'm not sure which compatability APIs are off limits and which ones aren't.

Basically, dealing with long paths on windows is a huge pain in the ass.

Maybe go does that stuff internally which is why the_platinum_searcher isn't affected?

**Comment by JFLarvoire** [comment]
This is fixed in the Windows build that I posted there:
https://github.com/JFLarvoire/the_silver_searcher/releases/download/1.0.2-Windows/ag.zip
(Internally using the \\\\?\\, etc, workarounds that @mitchblank mentioned.)

**Comment by JFLarvoire** [comment]
I've posted a new 2.0.0 release with this fix.  
The generic link for this new release, and all future releases is:  
https://github.com/JFLarvoire/the_silver_searcher/releases
